File: backend/services/reporting/app/indexer/utils_test.go
// Copyright 2023 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.

package indexer

import (
	"testing"

	"github.com/mendersoftware/mender-server/services/reporting/model"
	"github.com/stretchr/testify/assert"
)

func TestGroupJobsIntoTenantActionIDs(t *testing.T) {
	jobs := []model.Job{
		{
			Action:   model.ActionReindex,
			TenantID: "t1",
			DeviceID: "d1",
			Service:  model.ServiceInventory,
		},
		{
			Action:   model.ActionReindex,
			TenantID: "t1",
			DeviceID: "d1",
			Service:  model.ServiceDeviceauth,
		},
		{
			Action:   model.ActionReindex,
			TenantID: "t1",
			DeviceID: "d2",
			Service:  model.ServiceInventory,
		},
		{
			Action:   model.ActionReindex,
			TenantID: "t2",
			DeviceID: "d1",
			Service:  model.ServiceInventory,
		},
		{
			Action:   model.ActionReindexDeployment,
			TenantID: "t2",
			ID:       "d1",
			Service:  model.ServiceInventory,
		},
		{
			Action:   model.ActionReindexDeployment,
			TenantID: "t2",
			ID:       "d2",
			Service:  model.ServiceInventory,
		},
	}

	tenantActionIDs := groupJobsIntoTenantActionIDs(jobs)
	expected := TenantActionIDs{
		"t1": ActionIDs{
			model.ActionReindex: {
				"d1": true,
				"d2": true,
			},
		},
		"t2": ActionIDs{
			model.ActionReindex: {
				"d1": true,
			},
			model.ActionReindexDeployment: {
				"d1": true,
				"d2": true,
			},
		},
	}

	assert.Equal(t, expected, tenantActionIDs)
}

File: frontend/src/js/components/subscription/utils.test.ts
// Copyright 2025 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
import { parseSubscriptionPreview } from './utils';

const stripePreview = {
  id: 'upcoming_in_1Rabc',
  period_start: '2025-07-28T10:39:43Z',
  period_end: '2025-07-28T10:39:43Z',
  total: 3200,
  currency: 'usd',
  lines: [
    {
      description: '50 device × Mender Basic (Tier 1 at $0.00 / month)',
      amount: 0,
      currency: 'usd',
      quantity: 50,
      price_id: 'price_1abc'
    },
    {
      description: 'Mender Basic (Tier 1 at $32.00 / month)',
      amount: 3200,
      currency: 'usd',
      quantity: 0,
      price_id: 'price_1abc'
    },
    {
      description: '50 device × Mender Configure (Tier 1 at $0.00 / month)',
      amount: 0,
      currency: 'usd',
      quantity: 50,
      price_id: 'price_1PYl8wFlFfXikjZVIlqZ6zYr'
    },
    {
      description: 'Mender Configure (Tier 1 at $10.00 / month)',
      amount: 1000,
      currency: 'usd',
      quantity: 0,
      price_id: 'price_1PYl8wFlFfXikjZVIlqZ6zYr'
    },
    {
      description: '50 device × Mender Troubleshoot (Tier 1 at $0.00 / month)',
      amount: 0,
      currency: 'usd',
      quantity: 50,
      price_id: 'price_1PYkxZFlFfXikjZVJYRaUPld'
    },
    {
      description: 'Mender Troubleshoot (Tier 1 at $20.00 / month)',
      amount: 2000,
      currency: 'usd',
      quantity: 0,
      price_id: 'price_1PYkxZFlFfXikjZVJYRaUPld'
    }
  ]
};
describe('subscription utils', () => {
  it('stripe subscription preview', async () => {
    const result = parseSubscriptionPreview(stripePreview.lines, 'basic');
    expect(result.plan).toEqual(3200);
    expect(result.addons.configure).toEqual(1000);
    expect(result.addons.troubleshoot).toEqual(2000);
  });
});
